cypress/e2e/main.cy.ts

```diff
@@ -85,31 +87,29 @@ describe('Visual Regression Example', () => {
     } else {
       cy.visit('./cypress/web/05.html')
       cy.get('H1').contains('none').should('exist')
-      // @ts-ignore TODO
-      cy.compareSnapshot('foo', 0.02).should((comparisonResult: ComparisonResult) => {
-        expect(comparisonResult.error).to.be.undefined
-        expect(comparisonResult.percentage).to.be.below(0.02)
+      cy.compareSnapshot('foo', 0.02).should((result: VisualRegressionResult) => {
+        expect(result.error).to.be.undefined
+        expect(result.percentage).to.be.below(0.02)
       })
+      // @ts-ignore TODO type error
```

Comment:
as a reminder, it is best to use `@ts-expect-error` instead of  `@ts-ignore`, so we will not forget about those comments if we fix the type error.

The rule is set right now to warning because we have an issue when using different versions of cypress but ideally in the future we should have this rule set to error 

https://typescript-eslint.io/rules/ban-ts-comment/

Reply:
Will keep that in mind. But I think I managed to resolve all those type issues (except 1 `@ts` and 2 `eslint`)